### PR TITLE
Feature: Update checkbox behavior in details layout to replace icon

### PR DESCRIPTION
**Pull request by chingucoding**
This PR updates the behavior of the selection checkboxes in Details layout. Insteaed of listening to the app settings, now the "select all checkbox" will always be visible and the rows checkboxes will replace the icon if the user is hovering over the row or any row is selected.

**Validation**
How did you test these changes?
- [X] Built and ran the app
- [X] Tested the changes for accessibility

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Comment by yaira2** [comment]
@chingucoding can you fix the merge conflicts and update from `main`?

**Comment by yaira2** [comment]
@chingucoding it looks like the xaml got messed up in the merge, can you take a look?

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml [hunk 1]
```diff
@@ -547,7 +550,7 @@
 								Visibility="{x:Bind ColumnsViewModel.StatusColumn.Visibility, Mode=OneWay}" />
 
 							<controls:GridSplitter
-								Grid.Column="20"
+								Grid.Column="19"
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:553]: Does changing the columns affect resizing them?

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml [hunk 2]
```diff
@@ -561,247 +565,272 @@
...
+									IsRightTapEnabled="True"
+									Loaded="StackPanel_Loaded"
+									Orientation="Horizontal"
+									ToolTipService.ToolTip="{x:Bind ItemTooltipText, Mode=OneWay}">
+									<Grid
+										Width="24"
 										HorizontalAlignment="Left"
-										VerticalAlignment="Bottom"
-										x:Load="{x:Bind IsShortcut}">
-										<uc:OpacityIcon Style="{StaticResource ColorIconShortcut}" />
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:637]: Was this removed on purpose?

Comment (chingucoding) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:637]: Looks like a bad merge, fixed now.

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml (hunk 1 above)
Comment (chingucoding) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:554]: Given that we removed a column, my guess would be no and some testing also indicate resizing still works.

File: src/Files.App/Views/Settings/FoldersPage.xaml [hunk 3]
```diff
@@ -238,18 +238,6 @@
 					Style="{StaticResource RightAlignedToggleSwitchStyle}" />
 			</local:SettingsBlockControl>
 
-			<!--  Show checkboxes next to file names  -->
-			<local:SettingsBlockControl Title="{helpers:ResourceString Name=SettingsCheckboxesToSelectItems}" HorizontalAlignment="Stretch">
```
Comment (yaira2) [inline src/Files.App/Views/Settings/FoldersPage.xaml:242]: Can you remove the string?

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:555]: Double clicking a grid splitter to resize the column doesn't seem to work. Same with the size all to fit option.

Comment (chingucoding) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:555]: Are those behaviors currently present in the store version? I'm not sure what "double clickign" the grid splitter is supposed to do. The behavior for both things you mentioned seems to be the same compared to the store version 🤔

Comment (yaira2) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:555]: Compare with `main`

Comment (yaira2) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:555]: This behavior was broken in the store version and fixed in `main`

Comment (chingucoding) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:555]: I think I fixed this now. The whole column logic is a real pain to work with though...

Comment (yaira2) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:555]: You're telling me 😂

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**